rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py

```diff
@@ -522,6 +522,35 @@ def test_exec(config, ssh_con):
                                     raise TestExecError(
                                         "Bucket stats for non-existent bucket should return failure (2) or ENOENT."
                                     )
+                    if config.test_bi_purge is True:
+                        cmd = "radosgw-admin bucket stats --bucket=%s" % bucket.name
+                        out = utils.exec_shell_cmd(cmd)
+                        json_doc = json.loads(out)
+                        bucket_id = json_doc["id"]
+                        log.info(
+                            "Remove the bucket via bucket rm and --bypass-gc option"
+                        )
+                        utils.exec_shell_cmd(
+                            f"radosgw-admin bucket rm --bucket={bucket.name} --bypass-gc --purge-objects"
+                        )
+                        log.info(f"Do bi list for bucket {bucket.name}")
+                        utils.exec_shell_cmd(
+                            f"radosgw-admin bi list --bucket={bucket.name} --bucket-id={bucket_id}"
+                        )
+                        log.info(f"Do bi purge for bucket {bucket.name}")
+                        utils.exec_shell_cmd(
+                            f"radosgw-admin bi purge --bucket={bucket.name} --bucket-id={bucket_id}"
+                        )
+                        log.info(
+                            f"Do bi list for bucket {bucket.name} again, it should be empty and return 2"
+                        )
+                        cmd = f"radosgw-admin bi list --bucket={bucket.name} --bucket-id={bucket_id}"
+                        ec, _ = sp.getstatusoutput(cmd)
+                        if ec != 2:
+                            raise TestExecError(
```

Comment:
Please replace this with `raise AssertionError(` since `raise TestExecError(` wont break exec flow

Reply:
@udaysk23 Has there been changes for TestExecError recently?

Reply:
For me it works with TestExec as well.

2022-09-05 07:45:23,747 ERROR: bi list after bi purge is not empty, it's a test failure.
2022-09-05 07:45:23,748 ERROR: Traceback (most recent call last):
  File "test_Mbuckets_with_Nobjects.py", line 693, in <module>
    test_exec(config, ssh_con)
  File "test_Mbuckets_with_Nobjects.py", line 551, in test_exec
    "bi list after bi purge is not empty, it's a test failure."
v2.lib.exceptions.TestExecError: bi list after bi purge is not empty, it's a test failure.

2022-09-05 07:45:23,748 INFO: status: test failed

Reply:
Ahh my bad... I was mistakenly referring to `AddTestInfo` class